config/src/main/java/no/sysco/middleware/kafka/clients/interceptor/config/ConfigHarvesterInterceptor.java

```diff
@@ -0,0 +1,146 @@
+package no.sysco.middleware.kafka.clients.interceptor.config;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerInterceptor;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerInterceptor;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.streams.StreamsConfig;
+
+/**
+ * Kafka interceptor for harvesting and storing Producer/Consumer/KafkaStreams user provided configs
+ *
+ * @author 100tsa
+ * @param <K>
+ * @param <V>
+ */
+public class ConfigHarvesterInterceptor<K, V> implements ProducerInterceptor<K, V>, ConsumerInterceptor<K, V> {
+
+    private static final String TOPIC_NAME = "__clients";

```

Comment:
Does not `__client_configs` make more sense for this use-case? Also, we should think in a way to make this configurable, using this name as default.

Reply:
fixed